### PR TITLE
Add support for menuItem extension point targetMode

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Follow up to #4875
- Adds a `targetMode` to the button extension point that matches the behavior of the quick bar and context menu
- 🎉  Adding test coverage of the menuItemExtension and triggerExtension extensionPoint definition files
- Deprecates the `readerSelector` option for menuItem extension points

## Discussion

- Leaving the "do not merge" label on so we can discuss if this is the right approach
- The button extension point has support for a `readerSelector` property that's not exposed in the Page Editor
- Now that we've added support for element refs and the Traverse Elements brick, it's not necessary
- targetMode defaults to `document`. That's different than the behavior of the context menu
- Trigger's "targetMode" is a bit different due to event bubbling. It's "eventTarget" for the exact element that caused the event, and "root" for closest element that matches the "rootSelector". There's currently no "document" setting for trigger's root mode

## Demo

- https://www.loom.com/share/eccaaaf59426402da7ca7e495b201b64

## Checklist

- [x] Add tests
- [x] Designate a primary reviewer: @BLoe

### CODE REVIEW

File: src/blocks/registry.ts [hunk 1]
```diff
@@ -78,11 +78,6 @@ export class BlocksRegistry extends BaseRegistry<RegistryId, IBlock> {
     return this.typeCachePromise;
   }
 
-  async lookupTyped(id: RegistryId): Promise<TypedBlock | undefined> {
```
Comment (twschiller) [inline src/blocks/registry.ts:81]: Unused code

File: package.json [hunk 2]
```diff
@@ -235,6 +235,7 @@
     "jest-extended": "^3.1.0",
     "jest-webextension-mock": "^3.8.7",
     "jsdom": "^20.0.0",
+    "jsdom-testing-mocks": "^1.6.0",
```
Comment (twschiller) [inline package.json:238]: Required for kbar animation

File: src/extensionPoints/menuItemExtension.ts [hunk 3]
```diff
@@ -916,6 +957,16 @@ export class RemoteMenuItemExtensionPoint extends MenuItemExtensionPoint {
       return $elements.get(0);
     }
 
+    if (this._definition.targetMode === "eventTarget") {
+      if ($buttonElement == null) {
+        throw new BusinessError(
+          "eventTarget not supported for buttons with dynamic captions"
```
Comment (BLoe) [inline src/extensionPoints/menuItemExtension.ts:963]: Just curious, why does a null `$buttonElement` mean that it has a dynamic caption?

**Review by BLoe — APPROVED** [review]
Just had a question, but LGTM 👍

File: src/extensionPoints/menuItemExtension.ts (hunk 3 above)
Comment (twschiller) [reply src/extensionPoints/menuItemExtension.ts:963]: That happens to be the code path where the method gets called with a null for $buttonElement. Before merging I'll look at clarifying where the error is thrown/the comments

**Pull request merged**